src/test/java/com/amihaiemil/zold/mock/ArrayPayloadOf.java

```diff
@@ -54,10 +54,10 @@
      * @param request The http request
      * @throws IllegalStateException if the request's payload cannot be read
      */
-    public ArrayPayloadOf(final HttpRequest request) {
-        try (JsonReader reader = Json.createReader(
-            ((HttpEntityEnclosingRequest) request).getEntity().getContent())) {
-            if (request instanceof HttpEntityEnclosingRequest) {
+    public ArrayPayloadOf(final ClassicHttpRequest request) {
+        final HttpEntity entity = request.getEntity();
+        try (JsonReader reader = Json.createReader(entity.getContent())) {
+            if (entity != null) {
```

Comment:
@wholegroup , a line before `entity.getContent()` has been invoked already. Could you please clarify the reason this check has to be present for? In case it has to be, please do it earlier.

Reply:
fixed